runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -498,6 +508,37 @@ private boolean hasMultipleOutputs(Map<TupleTag<?>, PCollection<?>> outputs) {
         return outputs.size() > 1;
       }
 
+      /**
+       * Filter out obsolete, unused output tags except for {@code mainTag}.
+       *
+       * <p>This can help to avoid unnecessary caching in case of multiple outputs if only {@code
+       * mainTag} is consumed.
+       */
+      private Map<TupleTag<?>, PCollection<?>> skipObsoleteOutputs(
```

Comment:
What is "obsolete" output in this case?

Reply:
It means obsolete as no longer in use / unused. I did use the term because it is used in Structured streaming runner and I decided to keep is consistent:
https://github.com/apache/beam/blob/v2.61.0/runners/spark/3/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/batch/ParDoTranslatorBatch.java#L119

I am happy to change it if you feel it more appropriate.

Reply:
I'd say unused/unconsumed would be more clear. We could change it on both places?